components/service/glean/src/main/java/mozilla/components/service/glean/config/Configuration.kt

```diff
@@ -0,0 +1,18 @@
+package mozilla.components.service.glean.config
+
+/**
+ * The Configuration class describes how to configure the Glean.
+ *
+ * @property serverEndpoint the server pings are sent to
+ * @property userAgent the user agent used when sending pings
+ * @property connectionTimeout the timeout, in milliseconds, to use when connecting to
+ *           the [serverEndpoint]
+ * @property readTimeout the timeout, in milliseconds, to use when connecting to
+ *           the [serverEndpoint]
+ */
+public data class Configuration(
+    val serverEndpoint: String,
+    val userAgent: String,
+    val connectionTimeout: Int,
+    val readTimeout: Int
+)
```

Comment:
I wonder if `DefaultConfiguration` is really needed. You can define default values for the properties here. It will allow the app to override specific ones if needed. But I guess most of the time the default will be good enough here?

E.g.:

```Kotlin
data class Configuration(
    val serverEndpoint: String = "https://incoming.telemetry.mozilla.org",
    val userAgent: String = "Glean/1.0 (Android)",
    val connectionTimeout: Int = 1000,
    val readTimeout = 30000
}
```

Also note that `public` is redundant here. By default everything is public in Kotlin.